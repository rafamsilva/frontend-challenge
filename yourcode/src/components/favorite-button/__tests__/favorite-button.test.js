import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';

import FavoriteButton from '../';

describe('FavoriteButton Component', () => {
  let container;

  beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
  });

  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  describe('Render', () => {
    it('should render without crashing', () => {
      act(() => {
        render(<FavoriteButton />, container);
      });
    });

    it('should render liked button', () => {
      act(() => {
        render(<FavoriteButton active={true} />, container);
      });

      expect(container.querySelector("[data-testid='liked-icon']")).toBeDefined();
      expect(container.querySelector("[data-testid='unliked-icon']")).toBeNull();

      expect(container.querySelector("[data-testid='text']").textContent).toBe('Added');
    });

    it('should render unliked button', () => {
      act(() => {
        render(<FavoriteButton />, container);
      });

      expect(container.querySelector("[data-testid='unliked-icon']")).toBeDefined();
      expect(container.querySelector("[data-testid='liked-icon']")).toBeNull();

      expect(container.querySelector("[data-testid='text']").textContent).toBe('Add to favourites');
    });
  });

  describe('Snapshot', () => {
    it('should match snapshot active', () => {
      act(() => {
        render(<FavoriteButton active={true} />, container);
      });

      expect(pretty(container.innerHTML)).toMatchSnapshot();
    });

    it('should match snapshot not active', () => {
      act(() => {
        render(<FavoriteButton />, container);
      });

      expect(pretty(container.innerHTML)).toMatchSnapshot();
    });
  });
});
