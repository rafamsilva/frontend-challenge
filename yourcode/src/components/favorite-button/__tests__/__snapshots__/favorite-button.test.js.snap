// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteButton Component Snapshot should match snapshot active 1`] = `"<span class=\\"sc-AxjAm eNFStq\\"><span class=\\"sc-AxirZ blCCDh\\"><svg alt=\\"Unlike\\" title=\\"Unlike\\" data-testid=\\"liked-icon\\">icon-heart-full.svg</svg></span><span data-testid=\\"text\\" class=\\"sc-AxiKw efDhAh\\">Added</span></span>"`;

exports[`FavoriteButton Component Snapshot should match snapshot not active 1`] = `"<span class=\\"sc-AxjAm gEBWDb\\"><span class=\\"sc-AxirZ blCCDh\\"><svg alt=\\"Like\\" title=\\"Like\\" data-testid=\\"unliked-icon\\">icon-heart-grey.svg</svg></span><span data-testid=\\"text\\" class=\\"sc-AxiKw hpNIQS\\">Add to favourites</span></span>"`;
